src/GraphQL/DI/IGraphQLBuilder.cs

```diff
@@ -0,0 +1,41 @@
+using System;
+
+namespace GraphQL.DI
+{
+    /// <summary>
+    /// An interface for configuring GraphQL.NET services.
+    /// </summary>
+    public interface IGraphQLBuilder
+    {
+        /// <summary>
+        /// Registers the service of type <paramref name="serviceType"/> with the dependency injection provider.
+        /// </summary>
+        IGraphQLBuilder Register(Type serviceType, Type implementationType, ServiceLifetime serviceLifetime);
+
+        /// <inheritdoc cref="Register(Type, Type, ServiceLifetime)"/>
+        IGraphQLBuilder Register(Type serviceType, Func<IServiceProvider, object> implementationFactory, ServiceLifetime serviceLifetime);
+
+        /// <inheritdoc cref="Register(Type, Type, ServiceLifetime)"/>
+        IGraphQLBuilder Register(Type serviceType, object implementationInstance);
+
+        /// <summary>
+        /// Registers the service of type <paramref name="serviceType"/> with the dependency injection provider if a service
+        /// of the same type has not already been registered.
+        /// </summary>
+        IGraphQLBuilder TryRegister(Type serviceType, Type implementationType, ServiceLifetime serviceLifetime);
+
+        /// <inheritdoc cref="TryRegister(Type, Type, ServiceLifetime)"/>
+        IGraphQLBuilder TryRegister(Type serviceType, Func<IServiceProvider, object> implementationFactory, ServiceLifetime serviceLifetime);
+
+        /// <inheritdoc cref="TryRegister(Type, Type, ServiceLifetime)"/>
+        IGraphQLBuilder TryRegister(Type serviceType, object implementationInstance);
+
+        /// <summary>
+        /// Configures an options class of type <typeparamref name="TOptions"/>.
```

Comment:
"options class" here implies `Microsoft.Extensions.Options.IOptions<TOptions>`. Right?

Reply:
Not really... As I see from code, TOptions is any class and `TOptions` is injected, not `IOptions<TOptions>`.

In this method I feel some leaky abstraction to work with exactly `IOptions<T>`. I can not say what to do about it. Perhaps nothing.

Reply:
That’s why I did not import the options nuget package everywhere. (???)